### ISSUE TITLE
AI Agent feature

### ISSUE THREAD

**Issue by RaySinner** [continuedev/continue#2274]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

Currently, AI agents for writing full-cycle applications are gaining more and more popularity, and although they still cannot do anything truly worthwhile, I am sure that if you start moving in this direction, you will be able to remain market leaders at the moment when such agents can write decent solutions

**Comment by sestinj** [comment]
We have something in progress here :) aiming to focus on use cases that are concretely useful and well-scoped.

Will let you know soon as it's approaching release!

**Comment by itzhoujun** [comment]
Hi. Any update ?

**Comment by RaySinner** [comment]
The dudes have clearly fucked up the flash, now they are forced to trail behind, they are not even in the top ten, but they were once in first place. Everyone can disperse; their time is up.

![image](https://github.com/user-attachments/assets/caac595f-2685-4861-870a-9fd27beb7e7d)

**Comment by Symbolk** [comment]
No updates so... Are anyone interested in combining Continue and Cline? Especially integrating Cline agent to Continue?

**Comment by chezsmithy** [comment]
We are running both tools side by side. I've just been working on adding AWS profile support to cline to make it even easier to run side by side using Bedrock.

**Comment by Symbolk** [comment]
I started a PR https://github.com/continuedev/continue/pull/4216 for this, looking forward to discussion and contribution!

**Comment by Swipe4057** [comment]
@RaySinner

The screenshot showing the top API consumers by token count doesn't reflect the effectiveness of problem-solving, the quality of implemented agents, or the popularity of the application itself. Have you used Cline/Roo-Cline? These applications are oriented toward LLMs with long context, and even when solving a simple task, they consume nearly 120,000 tokens at once. Each agent move is added to the chat history and sent again, which is hugely inefficient, simply terrible, and interaction costs can be very high if it's an expensive API model, or if your local models on your own hardware don't support this. When the context approaches its limit, messages from history start being completely discarded, agents forget what they've already done and start doing it again, getting stuck in loops - there are no planning tools, memory, etc. For me, this top token consumption ranking reflects the inefficiency of Cline/Roo-Cline and its forks, of which there are many, so these are terrible extensions - in my personal opinion. The Continue community can implement a much better solution.

**Comment by RaySinner** [comment]
@Swipe4057 
I no longer use Continue completely switched to ROO-CLINE with a free VS Code LM provider I choose 5-10 million tokens per day by the Anthropic Sonnet 3.5 model for free, literally for zero dollars. So I don't understand what you mean. Continue was crap and hopelessly behind.

**Comment by Swipe4057** [comment]
@RaySinner You won't be able to spend that many tokens per day if free access is discontinued, as the costs would be too high. Moreover, there are many corporations where using external APIs is unacceptable from a security standpoint; no one within corporations would allow leaking internal company code to an API provider. You will need to deploy your own open-source models within the company on open-source inference engines. And these extensions generate meaningless overhead load and enormous context lengths. From this perspective, these extensions and their agent implementations are completely inefficient and pointless.

**Comment by RaySinner** [comment]
@Swipe4057 I do not see any problems locally expand some kind of open source model. The problem is that Continue decided that she knows better and f*ck up in her ideas about how the market will develop. I hope they will still pick up themselves and make a good competitive agent.

**Comment by ninapool** [comment]
Yeah I stopped using continue too because of the lack of an agents feature. It would be good if this PR was implemented though might make me consider going back

**Comment by pcuci** [comment]
Isn't https://hub.continue.dev/explore/assistants tackling exactly this desired agentic use-case?

**Comment by emahfoud** [comment]
It looks like there is a PR for this but it's not merged yet
https://github.com/continuedev/continue/pull/4216

**Comment by kolya-ay** [comment]
Wait guys, couldn't Continue use mcp server for agentic workflows (i.e. wrapper around aider for example)?

**Comment by RaySinner** [comment]
The agent’s functionality was added, many thanks are waiting for the further development of the project.
